src/nrtk_explorer/app/transforms.py

```diff
@@ -308,15 +308,16 @@ def _update_images(self, selected_ids):
     async def _set_source_images(self, selected_ids):
         # We need to yield twice for the self.state.loading_images=True to
         # commit to the trame state to show a spinner
-        await asyncio.sleep(0)
-        await asyncio.sleep(0)
+        # await asyncio.sleep(0)
+        # await asyncio.sleep(0)
         self._update_images(selected_ids)
         with self.state:
             self.state.loading_images = False
 
     def set_source_images(self, selected_ids):
         if len(selected_ids):
             self.state.loading_images = True
+            self.state.flush()
```

Comment:
Better to do (truly the same but look nicer)

```python
with self.state:
  self.state.loading_images = True
```

Reply:
Gotcha, thanks.  But, the spinner still does not appear without the `await asyncio.sleep(0)` calls.  Something else is blocking...